test/with_dummyserver/test_socketlevel.py

```diff
@@ -842,3 +842,56 @@ def test_empty_head_response_does_not_hang(self):
 
         # stream will use the read method here.
         self.assertEqual([], list(r.stream()))
+
+
+class TestChunkedTransfer(SocketDummyServerTestCase):
+    def _start_chunked_handler(self):
+        self.buffer = b''
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            while not self.buffer.endswith(b'\r\n0\r\n\r\n'):
+                self.buffer += sock.recv(65536)
+
+            sock.send(
+               b'HTTP/1.1 200 OK\r\n'
+               b'Content-type: text/plain\r\n'
+               b'Content-Length: 0\r\n'
+               b'\r\n')
+            sock.close()
+
+        self._start_server(socket_handler)
+
+    def test_chunks(self):
+        self._start_chunked_handler()
+        chunks = ['foo', 'bar', '', 'bazzzzzz']
+        pool = HTTPConnectionPool(self.host, self.port, retries=False)
+        r = pool.urlopen('GET', '/', chunks, headers=dict(DNT='1'), chunked=True)
+
+        self.assertTrue(b'Transfer-Encoding' in self.buffer)
```

Comment:
We can be a bit more aggressive here with this check: we should look for the exact header line and confirm it's well formed. `b'\r\nTransfer-Encoding: chunked\r\n`.